### ISSUE TITLE
ComponentLifeCycle doesn't seem to be executed on second navigation

### ISSUE THREAD

**Issue by isaaclem** [react-navigation/react-navigation#4866]
### Current Behavior

I've implemented `createBottomTabNavigator` with two `Screen/Component` as below
```
const MainNavigator = createBottomTabNavigator({
  auth: { screen: LoginScreen },
  main: {
    screen: createBottomTabNavigator({
      summary: { 
        screen: SummaryScreen,
        navigationOptions: {
          tabBarLabel: labelSummary,
          tabBarIcon: ({ tintColor }) => <Icon type='MaterialIcons' name='speaker-notes' style={{ color: tintColor, fontSize: 25 }} />
        }, 
      }
    }, {
      tabBarPosition: 'bottom',
      tabBarOptions: {
        labelStyle: { fontSize: 12 },
        activeTintColor: '#F73D65',
        pressColor: 'yellow',
        style: {
          // backgroundColor: 'yellow'
        }
      }
    })
  }
}, {
  navigationOptions: {
    tabBarVisible: false
  },
  lazy: true
});
```

On first load, `LoginScreen` will be rendered and using `props.navigation.navigate('summary');` to navigate to `SummaryScreen`, and based on certain logic, I'll kick them back to LoginScreen again using `this.props.navigation.navigate('auth');`, to ask user to login again. 

Now user going thru same flow try to login and hits `props.navigation.navigate('summary');`. 
`SummaryScreen` does rendered however, `constructor or componentWillMount, componentDidMount` doesn't seems to execute at all?

In conclusion, for the first time, when we navigate from screen1 to screen2, screen2's componentLifeCycle did executed but if we navigate back to screen1 and renavigate to screen2 again, componentLifeCycle didnt seems to get executed?

### Expected Behavior

ComponentLifeCycle methods gets executed regardless how many times we navigate users

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | ^2.11.2
| react-native     |  0.56.0
| npm or yarn      | yarn

**Comment by brentvatne** [comment]
the tabs are lazily rendered but once they are rendered they aren't disposed of, so if you go back to it then it will not re-render from scratch. you can use switch navigator if you want that behavior

**Comment by isaaclem** [comment]
@brentvatne Is there by any chance we can force it to rerender manually?

**Comment by eyalabadi98** [comment]
@isaaclem I am having the same issue. When a user logs in, it fetches their user status, but the navigator renders as soon as the app opens and its not getting the response on time.

I tried multiple ways to rerender like setstate but nothing.
Thanks

**Comment by brentvatne** [comment]
this is what https://reactnavigation.org/docs/en/with-navigation-focus.html is for
